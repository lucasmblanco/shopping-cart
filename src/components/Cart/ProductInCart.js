import React, { useEffect, useState } from 'react';
import {HiTrash} from 'react-icons/hi'

export default function ProductInCart({product, deleteProduct, addQuantity}) {

  //  const [ quantity, dispatch ] = useReducer(reducer, product.quantity); 
  //  const [price, dispatch2] = useReducer(reducer, product.price); 
    const [ limit, setLimit ] = useState(false)
    
    useEffect(() => {
        product.quantity <= 1 ? setLimit(true) : setLimit(false)
      }, 
      [product.quantity])

  return (
   <>
    <div className=' flex justify-around text-xs text-left text-white-accent h-16 '>
        <div className='grid content-center text-[8px] w-[10rem]  text-white-accent  p-4'>{product.title}</div>
        <div className='w-[3rem] grid grid-cols-cart-quantity bg-blue-accent m-1 rounded-md'> 
            <div className='col-span-2 justify-self-center self-end'>{product.quantity}</div>
            <button className='text-green-500 transition-all duration-500 hover:text-green-300'onClick={()=> {addQuantity(product.id, 'decrement')}} disabled={limit}>-</button>
            <button className='text-red-500 transition-all duration-500 hover:text-red-300' onClick={()=> {addQuantity(product.id, 'increment')}}>+</button>
     </div>
        <div className=' grid content-center justify-center text-base  w-[5rem] transition-all duration-500'>$ {product.price * product.quantity}</div>
        <div className="flex justify-center rounded-full text-red-accent transition-all duration-500 hover:text-light-red-accent p-4"><button onClick={() => deleteProduct(product.id)}><HiTrash /></button></div>
    </div>
    <hr></hr>
   </>
  )
}
