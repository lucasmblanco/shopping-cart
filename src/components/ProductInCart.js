import React, { useEffect, useState } from 'react'

export default function ProductInCart({product, deleteProduct, addQuantity}) {

  //  const [ quantity, dispatch ] = useReducer(reducer, product.quantity); 
  //  const [price, dispatch2] = useReducer(reducer, product.price); 
    const [ limit, setLimit ] = useState(false)
    
    useEffect(() => {
        product.quantity <= 1 ? setLimit(true) : setLimit(false)
      }, 
      [product.quantity])

  return (
    <div className='text-xs text-left p-3 flex'>
        <div>{product.name}</div>
        <div> 
            <button onClick={()=> {addQuantity(product.id, 'decrement')}} disabled={limit}>-</button>
            <div>{product.quantity}</div>
            <button onClick={()=> {addQuantity(product.id, 'increment')}}>+</button>
     </div>
        <div>{product.totalPrice}</div>
        <button onClick={() => deleteProduct(product.id)}>delete</button>
    </div>
  )
}
