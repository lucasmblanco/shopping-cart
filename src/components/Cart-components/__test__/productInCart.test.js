import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import ProductInCart from '../ProductInCart';

const MOCK_PRODUCT = {
  id: 1,
  title: 'MockProduct',
  quantity: 1,
  price: 1,
};

const addFunction = jest.fn();
const deleteFunction = jest.fn();

it('should render component', () => {
  const { asFragment } = render(
    <ProductInCart
      product={MOCK_PRODUCT}
      deleteProduct={deleteFunction}
      addQuantity={addFunction}
    />,
  );

  expect(asFragment()).toMatchSnapshot();
});

it('should display correct amount of quantity', async () => {
  render(
    <ProductInCart
      product={MOCK_PRODUCT}
      deleteProduct={deleteFunction}
      addQuantity={addFunction}
    />,
  );

  const incrementQuantity = screen.getByRole('button', { name: '+' });
  //const decrementQuantity = screen.getByTestId('decrement-quantity');

  await userEvent.click(incrementQuantity);
  //userEvent.click(decrementQuantity);

  expect(addFunction).toHaveBeenCalled();
});

it('should delete product', async () => {
  render(
    <ProductInCart
      product={MOCK_PRODUCT}
      deleteProduct={deleteFunction}
      addQuantity={addFunction}
    />,
  );

  const deleteProduct = screen.getByTestId('delete-product');

  await userEvent.click(deleteProduct);

  expect(deleteFunction).toHaveBeenCalled();
});
