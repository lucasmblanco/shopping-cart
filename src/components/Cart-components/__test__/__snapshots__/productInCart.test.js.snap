// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<DocumentFragment>
  <div
    class=" flex justify-around text-xs text-left text-white-accent h-16 "
    data-testid="product-in-cart"
  >
    <div
      class="grid content-center text-[8px] w-[10rem]  text-white-accent  p-4"
    >
      MockProduct
    </div>
    <div
      class="w-[3rem] grid grid-cols-cart-quantity bg-blue-accent m-1 rounded-md"
    >
      <div
        class="col-span-2 justify-self-center self-end"
        data-testid="product-quantity"
      >
        1
      </div>
      <button
        class="text-green-500 transition-all duration-500 hover:text-green-300"
        data-testid="decrement-quantity"
        disabled=""
      >
        -
      </button>
      <button
        class="text-red-500 transition-all duration-500 hover:text-red-300"
        data-testid="increment-quantity"
      >
        +
      </button>
    </div>
    <div
      class=" grid content-center justify-center text-base  w-[5rem] transition-all duration-500"
    >
      $ 1
    </div>
    <div
      class="flex justify-center rounded-full text-red-accent transition-all duration-500 hover:text-light-red-accent p-4"
    >
      <button
        data-testid="delete-product"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9 2a1 1 0 00-.894.553L7.382 4H4a1 1 0 000 2v10a2 2 0 002 2h8a2 2 0 002-2V6a1 1 0 100-2h-3.382l-.724-1.447A1 1 0 0011 2H9zM7 8a1 1 0 012 0v6a1 1 0 11-2 0V8zm5-1a1 1 0 00-1 1v6a1 1 0 102 0V8a1 1 0 00-1-1z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
  <hr />
</DocumentFragment>
`;
